dev/src/rate-limiter.ts

```diff
@@ -90,6 +90,7 @@ export class RateLimiter {
     numOperations: number,
     requestTimeMillis = Date.now()
   ): number {
+    this.refillTokens(requestTimeMillis);
```

Comment:
Hm. This method read to me as "let me check when I can make the next request, but there won't be any side effects". Can you refactor this to not have side effects?

Reply:
To give an accurate estimate of how many ms are required to make the next request, we have to refill the tokens when this method is called. Otherwise, `getNextRequestDelayMs()` can return an incorrect value.
 
Without `refillTokens()` in `getNextRequestDelayMs()`:
```
tryMakeRequest(500, time(0 seconds)) // success
getNextRequestDelay(500, time(1 second)) // there are no availableTokens without refill, incorrectly returns 1000ms.
```

Since the token refill logic takes into account the last refill time, calling `refillTokens()` again as part of `tryMakeRequest()` won't have any unintended side effects.

Reply:
That's fair. It seems like this could potentially used by simplified to not store the request time at all and to always refill tokens by default.